versionsupport_common/src/main/java/com/tomkeuper/bedwars/listeners/ItemDropPickListener.java

```diff
@@ -56,8 +52,12 @@ public void onDrop(PlayerDropItemEvent e){
     public static class PlayerPickup implements Listener {
         @SuppressWarnings("deprecation")
         @EventHandler
-        public void onDrop(PlayerPickupItemEvent e){
-            if (managePickup(e.getItem(), e.getPlayer())) e.setCancelled(true);
+        public void onPickUp(PlayerPickupItemEvent e) {
+            if (managePickup(e.getItem(), e.getPlayer(), getItemsAround(e.getItem()).size())) {
+                e.setCancelled(true);
+                return;
+            }
+            manageGeneratorPickUp(e, e.getPlayer(), e.getItem(), getItemsAround(e.getItem()));
```

Comment:
Why do we have to split up the managePickup and the manageGeneratorPickup? Arent they always used together? If thats the case then the code can be kept the same but with the addition of the improved managePickup function